logstash-core/src/main/java/org/logstash/ackedqueue/Batch.java

```diff
@@ -21,6 +22,17 @@ public Batch(List<Queueable> elements, LongVector seqNums, Queue q) {
         this.closed = new AtomicBoolean(false);
     }
 
+    public Batch(SequencedList<byte[]> serialized, Queue q) {
+        final List<byte[]> serializedElements = serialized.getElements();
+        this.elements = new ArrayList<>(serializedElements.size());
+        for (final byte[] element : serializedElements) {
+            this.elements.add(q.deserialize(element));
+        }
+        this.seqNums = serialized.getSeqNums();
```

Comment:
@colinsurprenant I wouldn't duplicate primary constructors here. Just extract the deserialisation loop to a method `private static List<Queueable> deserializeElements(Queue q, Collection<byte[]> c)` and make the secondary constructor simply call:
```
this(deserialize(q, serialized.getElements()), serialized.getSeqNums(), q);
```

... or just drop the `public Batch(List<Queueable> elements, LongVector seqNums, Queue q)` (maybe it's unnecessary now) ?

Other than that, looks fine :)

Reply:
+1